daemon/src/device.rs

```diff
@@ -60,6 +60,10 @@ pub struct Device<'a> {
     global_events: Sender<EventTriggers>,
 
     last_sample_error: Option<String>,
+
+    // Tap-tempo configuration and state
+    tap_tempo_window: Duration,
+    tap_states: HashMap<EffectBankPresets, TapTempoState>,
```

Comment:
Why is this HashMapped specifically to the preset? It's a 'general purpose' handler that doesn't have a relation to the button being pressed, we can simply have one TapTempoState and reset it when the preset changes.